pkg/server/operational-handler/operational_handler.go

```diff
@@ -0,0 +1,254 @@
+package operationalhandler
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/Workiva/go-datastructures/queue"
+	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+// Value represents a gNMI value with path and data.
+// This is a minimal version to avoid importing the full sonic proto package.
+type Value struct {
+	Path      *gnmipb.Path
+	Value     *gnmipb.TypedValue
+	Timestamp int64
+}
+
+// Client is the minimal interface required for gNMI handlers.
+// This avoids importing the full sonic_data_client package.
+type Client interface {
+	StreamRun(q *queue.PriorityQueue, stop chan struct{}, w *sync.WaitGroup, subscribe *gnmipb.SubscriptionList)
+	PollRun(q *queue.PriorityQueue, poll chan struct{}, w *sync.WaitGroup, subscribe *gnmipb.SubscriptionList)
+	AppDBPollRun(q *queue.PriorityQueue, poll chan struct{}, w *sync.WaitGroup, subscribe *gnmipb.SubscriptionList)
+	OnceRun(q *queue.PriorityQueue, once chan struct{}, w *sync.WaitGroup, subscribe *gnmipb.SubscriptionList)
+	Get(w *sync.WaitGroup) ([]*Value, error)
+	Set(delete []*gnmipb.Path, replace []*gnmipb.Update, update []*gnmipb.Update) error
+	Capabilities() []gnmipb.ModelData
+	Close() error
+	FailedSend()
+}
+
+// OperationalHandler implements the Client interface for operational state gNMI queries.
+// It handles paths like /sonic/system/filesystem[path=*]/disk-space for disk space monitoring
+// and other operational data.
+type OperationalHandler struct {
+	prefix       *gnmipb.Path
+	paths        []*gnmipb.Path
+	pathHandlers map[string]PathHandler
+	mu           sync.RWMutex
+}
+
+// PathHandler defines the interface for handling specific gNMI paths.
+type PathHandler interface {
+	// HandleGet processes a gNMI Get request for this path and returns the response data.
+	HandleGet(path *gnmipb.Path) ([]byte, error)
+
+	// SupportedPaths returns the list of paths this handler supports.
+	SupportedPaths() []string
+}
+
+// NewOperationalHandler creates a new OperationalHandler for the given paths and prefix.
+// It follows the same signature as other sonic-gnmi clients like NewNonDbClient.
+func NewOperationalHandler(paths []*gnmipb.Path, prefix *gnmipb.Path) (Client, error) {
+	handler := &OperationalHandler{
+		prefix:       prefix,
+		paths:        paths,
+		pathHandlers: make(map[string]PathHandler),
+	}
+
+	// Register path handlers
+	diskSpaceHandler := NewDiskSpaceHandler()
+	for _, supportedPath := range diskSpaceHandler.SupportedPaths() {
+		handler.pathHandlers[supportedPath] = diskSpaceHandler
+	}
+
+	// Validate that all requested paths are supported
+	for _, path := range paths {
+		pathStr := handler.pathToString(path)
+		if !handler.isPathSupported(pathStr) {
+			return nil, fmt.Errorf("unsupported path: %s", pathStr)
```

Comment:
Maybe we should return InvalidArgument or Unimplemented error code for unsupported path?